sale_order_type/models/sale.py

```diff
@@ -74,11 +75,34 @@ def onchange_type_id(self):
     @api.model
     def create(self, vals):
         if vals.get("name", "/") == "/" and vals.get("type_id"):
-            sale_type = self.env["sale.order.type"].browse(vals["type_id"])
-            if sale_type.sequence_id:
-                vals["name"] = sale_type.sequence_id.next_by_id()
+            vals["name"] = self._get_sequencing(vals["type_id"]) or self.name or "/"
         return super(SaleOrder, self).create(vals)
 
+    def write(self, vals):
+        """
+        Maintain proper sequencing when copying
+
+        """
+        if "type_id" not in vals:
+            return super(SaleOrder, self.write(vals))
+        type_id = vals.get("type_id")
+        # Store orders that type_id has changed
+        updated_orders = self.env["sale.order"]
+        for this in self:
+            # type_id remains the same.
+            if this.type_id.id == type_id:
+                continue
+            sequencing = self._get_sequencing(type_id)
```

Comment:
I don't like the name of this variable. It is a verb. Why not stick with next_sequence?